jccli/cli/group.py

```diff
@@ -14,29 +22,159 @@ def group(ctx):
 
 @group.command('create')
 @click.option('--name', "-n", required=True, type=str, help='Name of the group')
-@click.option('--type', "-t", required=True, type=click.Choice(['user', 'system']),
-              help='The type of group')
+@click.option('--user', 'type', flag_value='user_group')
+@click.option('--system', 'type', flag_value='system_group')
 @click.pass_context
 def create_group(ctx, name, type):
     """
     Create a group
     """
     api2 = JumpcloudApiV2(ctx.obj.get('key'))
+    logger: Logger = ctx.obj.get('logger')
+    if type is None:
+        logger.error('groups must have a type (either "user" or "system")')
+        sys.exit(1)
+    try:
+        api2.create_group(name, type)
+    except ApiException:
+        logger.error(f"API error (confirm that no group of type '{type}' and name '{name}' already exists)")
+        sys.exit(1)
+    logger.info(f"successfully created group: {name}")
+
+
+@group.command('get')
+@click.option('--name', '-n', required=True, type=str, help='Name of the group')
+@click.option('--user', 'type', flag_value='user_group')
+@click.option('--system', 'type', flag_value='system_group')
+@click.pass_context
+def get_group(ctx, name, type):
+    """
+    Get detail view of a group
+    """
+    api2 = JumpcloudApiV2(ctx.obj.get('key'))
     logger = ctx.obj.get('logger')
-    response = api2.create_group(name, type)
-    logger.info(f"{response}")
+    if type is None:
+        logger.error('groups must have a type (either "user" or "system")')
+        sys.exit(1)
+    response = api2.get_group(group_name=name, group_type=type)
+    serialized_response = json.dumps(response)
+    logger.info(f"{serialized_response}")
+
+
+@group.command('list')
+@click.option('--user', 'type', flag_value='user_group', help='Restrict to user groups only')
+@click.option('--system', 'type', flag_value='system_group', help='Restrict to system groups only')
+@click.pass_context
+def list_groups(ctx, type):
+    """
+    List groups
+    """
+    api2 = JumpcloudApiV2(ctx.obj.get('key'))
+    logger: Logger = ctx.obj.get('logger')
+    response = api2.get_groups(type=type)
+    serialized_response = json.dumps(response, indent=2)
+    logger.info(f"{serialized_response}")
 
 
 @group.command('delete')
```

Comment:
would be better if we changed delete to `remove` which will more closely match the powershell [Remove-JCUserGroup](https://github.com/TheJumpCloud/support/wiki/Remove-JCUserGroup) command

Reply:
Should the `create` command be changed to `add` to match the powershell equivalent, too?

Reply:
yes, i think we should but you can probably create an issue for this and handle it on a follow on PR.